### PR TITLE
feat(secrets): mask secrets findings in output

### PR DESCRIPTION
**Pull request by kopecs**
Currently masks everything past the first 20% of the finding in text
output, as well as metavariables by replacing the content with `*`s. This
is to avoid showing secrets in logs.

Test plan: existing pytest coverage.

### CODE REVIEW

**Review by akuhlens — APPROVED** [review]
This looks good.

**Review by akuhlens — COMMENTED** [review]
I am assuming that the test on the pro-engine side of things verify this works as expected.

**Comment by kopecs** [comment]
Yes, locally that was the case and you can see the update to the relevant OSS test too.

**Comment by kopecs** [comment]
Force push to rebase to test for pro compat.

File: src/osemgrep/reporting/Cli_json_output.ml [hunk 1]
```diff
@@ -389,6 +389,7 @@ let cli_match_of_core_match (hrules : Rule.hrules) (m : Out.core_match) :
             rule_message
       in
       (* message where the metavars have been interpolated *)
+      (* TODO(secrets): apply masking logic here *)
```
Comment (aryx) [inline src/osemgrep/reporting/Cli_json_output.ml:392]: When do you plan to port it to osemgrep?

Comment (kopecs) [reply src/osemgrep/reporting/Cli_json_output.ml:392]: Trying to find a good point to slot this in. I think I may be able late this week/early next.

**Pull request merged**